### PR TITLE
Removed link's underline from settings UI

### PR DESCRIPTION
**Pull request by simonhong**
fix https://github.com/brave/brave-browser/issues/26091

Screen Shot 2022-10-21 at 11 30 58 AM

Below upstream's change enabled underline.
https://chromium-review.googlesource.com/c/chromium/src/+/3855278 

WebUI: Decorate links with underline

This is part 1 of decorating all links and all buttons that look like links with the underline to help users better differentiate between normal text and actionable links. This CL updates links on Settings, Extensions, Management, Downloads, and Welcome, as well as updating a couple of global WebUI styles.

Resolves 

## Submitter Checklist:

- [x] I confirm that no security/privacy review [is needed](https://github.com/brave/brave-browser/wiki/Security-reviews), or that I have [requested](https://github.com/brave/security/issues/new/choose) one
- [x] There is a [ticket](https://github.com/brave/brave-browser/issues) for my issue
- [x] Used Github [auto-closing keywords](https://docs.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue) in the PR description above
- [x] Wrote a good [PR/commit description](https://google.github.io/eng-practices/review/developer/cl-descriptions.html)
- [ ] Squashed any review feedback or "fixup" commits before merge, so that history is a record of what happened in the repo, not your PR
- [x] Added appropriate labels (`QA/Yes` or `QA/No`; `release-notes/include` or `release-notes/exclude`; `OS/...`) to the associated issue
- [x] Checked the PR locally: `npm run test -- brave_browser_tests`, `npm run test -- brave_unit_tests`, `npm run lint`, `npm run gn_check`, `npm run tslint`
- [x] Ran `git rebase master` (if needed)

## Reviewer Checklist:

- [ ] A security review [is not needed](https://github.com/brave/brave-browser/wiki/Security-reviews), or a link to one is included in the PR description
- [ ] New files have MPL-2.0 license header
- [ ] Adequate test coverage exists to prevent regressions
- [ ] Major classes, functions and non-trivial code blocks are well-commented
- [ ] Changes in component dependencies are properly reflected in `gn`
- [ ] Code follows the [style guide](https://chromium.googlesource.com/chromium/src/+/HEAD/styleguide/c++/c++.md)
- [ ] Test plan is specified in PR before merging

## After-merge Checklist:

- [ ] The associated issue milestone is set to the smallest version that the
  changes has landed on
- [ ] All relevant documentation has been updated, for instance:
  - [ ] https://github.com/brave/brave-browser/wiki/Deviations-from-Chromium-(features-we-disable-or-remove)
  - [ ] https://github.com/brave/brave-browser/wiki/Proxy-redirected-URLs
  - [ ] https://github.com/brave/brave-browser/wiki/Fingerprinting-Protections
  - [ ] https://github.com/brave/brave-browser/wiki/Brave%E2%80%99s-Use-of-Referral-Codes
  - [ ] https://github.com/brave/brave-browser/wiki/Custom-Headers
  - [ ] https://github.com/brave/brave-browser/wiki/Web-Compatibility-Exceptions-in-Brave
  - [ ] https://github.com/brave/brave-browser/wiki/QA-Guide
  - [ ] https://github.com/brave/brave-browser/wiki/P3A

## Test Plan:

See the linked issue

### CODE REVIEW

**Review by emerick — APPROVED** [review]
LGTM

File: patches/chrome-browser-resources-settings-settings_shared.css.patch [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
++++ b/chrome/browser/resources/settings/settings_shared.css
+@@ -7,9 +7,10 @@
+  * #import=//resources/cr_elements/cr_shared_style.css.js
+  * #import=//resources/cr_elements/cr_shared_vars.css.js
+  * #import=//resources/cr_elements/search_highlight_style.css.js
++ * #import=//resources/page_specific/settings/br_settings_shared.css.js
+  * #import=./settings_vars.css.js
+ <if expr='not chromeos_ash'>
+- * #include=cr-shared-style search-highlight-style
++ * #include=cr-shared-style search-highlight-style br-settings-shared
```
Comment (goodov) [inline patches/chrome-browser-resources-settings-settings_shared.css.patch:13]: I believe our lines can be close to each other:
```
<if expr='not chromeos_ash'>
 * #import=//resources/page_specific/settings/br_settings_shared.css.js
 * #include=cr-shared-style search-highlight-style br-settings-shared
```

Comment (simonhong) [reply patches/chrome-browser-resources-settings-settings_shared.css.patch:11]: Moved.

**Review by goodov — APPROVED** [review]

**Comment by petemill** [comment]
Thanks for the fix. I think it's better if we put the css file in `brave/browser/resources/settings` and phase out page_specific, which was more of a convenience place to place overriding resources for chromium WebUIs where we haven't yet setup our places to inject modifications.

**Comment by kjozwiak** [comment]
Verification **PASSED** on `Win 11 x64` using the following build(s):

```
Brave | 1.47.14 Chromium: 107.0.5304.62 (Official Build) nightly (64-bit)
-- | --
Revision | 1eec40d3a5764881c92085aaee66d25075c159aa-refs/branch-heads/5304@{#942}
OS | Windows 11 Version 22H2 (Build 22621.675)
```

Went through the STR/Cases outlined via https://github.com/brave/brave-browser/issues/26091#issue-1412548867 and ensured that links weren't being underlined via `brave://settings` as per the following:

**`Example`** | **`Example`** | **`Example`**
---------------|-----------------|-----------------
underlinedFixed1 | underlinedFixed2 | underlinedFixed3

**Pull request merged**